pkg/parse/validate.go

```diff
@@ -0,0 +1,65 @@
+package parse
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	InterpolationBaseMock string = "x"
+	// Represents the syntax used by triggers to specify interpolation
+	InterpolationSyntax string = "$(" + InterpolationBaseMock + ")"
+	// The regex formatted version of the InterpolationSyntax
+	InterpolationRegex string = "\\$\\(" + InterpolationBaseMock + "\\)"
+	// The regex string specifying legal field names within data object
+	InterpolationFieldRegexString string = "[_a-zA-Z][_0-9a-zA-Z-]*"
+	// The base string to use within the InterpolationSyntax to specify the event object
+	EventInterpolationBase string = "event"
+	// The base string to use within the InterpolationSyntax to specify the params object
+	ParamsInterpolationBase string = "params"
+)
+
+// If template fields do not match this regex, their syntax is malformed.
+// This only a preliminary check and does not validate against the payload
+var eventRegex *regexp.Regexp
+
+// If param fields do not match this regex, their syntax is malformed or
+// they do not refer to param within the TriggerTemplate
+var paramsRegex *regexp.Regexp
+
+func init() {
+	eventRegex = regexp.MustCompile(createInterpolationRegex(EventInterpolationBase, true, true))
+	paramsRegex = regexp.MustCompile(createInterpolationRegex(ParamsInterpolationBase, false, false))
+}
+
+// Validates the interpolation used by TriggerBindings
+func ValidEventBinding(path string) bool {
+	return eventRegex.MatchString(path)
+}
+
+// Validates the interpolation used by TriggerTemplates
+func ValidTemplateBinding(path string) bool {
+	return paramsRegex.MatchString(path)
+}
+
+// Access paths created are separated using '.'
+// If baseAccess and recursiveAccess are both false, the desired regex will match 1 field deep
+func createInterpolationRegex(interpolationBase string, baseAccess bool, recursiveAccess bool) string {
+	var fieldAccessRegex string
+	if baseAccess && recursiveAccess {
+		fieldAccessRegex = fmt.Sprintf("(\\.%s)*", InterpolationFieldRegexString)
+	} else if recursiveAccess {
+		fieldAccessRegex = fmt.Sprintf("(\\.%s)+", InterpolationFieldRegexString)
+	} else if !baseAccess && !recursiveAccess {
+		fieldAccessRegex = fmt.Sprintf("(\\.%s){1}", InterpolationFieldRegexString)
+	}
+	baseAndFieldRegex := interpolationBase + fieldAccessRegex
+	interpolatedRegex := strings.Replace(InterpolationRegex, InterpolationBaseMock, baseAndFieldRegex, 1)
+	return interpolatedRegex
+}
```

Comment:
I think writing a test for this would probably feel a bit like a waste of time but I feel like it would be worth it 😇  - and make it easy to see the intended behaviour